tests/cpp17/stringify_util.h

```diff
@@ -74,19 +72,15 @@ std::string StringifyTableOrStructImpl(const FB &flatbuff,
   // deep copies of anything will be made.
   auto fields_pack = flatbuff.fields_pack();
   std::ostringstream oss;
-  auto add_field = [&](auto&& field_value, size_t index) {
```

Comment:
The auto-lambda is a nice feature, but not here. This code will produce exponential growth of the compiler's internal state. For each `FB` a compiler has to generate lambdas for every type T even if T has been already seen in the context of others FB. It would be better to extract a common templated method instead of lambda.

I've checked the generated assembler code, it is not perfect.
The compiler has to call all getters to return a result of `fields_pack()`.
I've tried to rewrite your code a little bit to improve performance (inline optimization).
This is the draft of code that generates acceptable assembler code: https://github.com/vglavnyy/flatbuffers/commit/fa325725d81232f091493cc167b7a3ad5325f796
Could your compare generated code?

Reply:
Note in this case it is OK that `fields_pack` is calling all getters because we need to use all of them anyway, and we are only calling `fields_pack` once per table/struct, not once per field (which would be inefficient).  We call it once, then iterate over it a few lines below.  Just wanted to make sure that we were on the same page about that, though if you think I am missing something let me know.

I'll will take a look at your implementation and compare generated code to see if the `get_field` method is able to get a single field without calling all getters.  But note that that might have very bad performance in a debug build.  For this Stringify case, we would likely still want to call `fields_pack`.

Reply:
Of course, in this function, we need to use all of them and both solutions do the same.
I've compared again both cases (external `std:get` and internal `std:get`) with MSVS Release build.

External  `std:get`:
- `call ?fields_pack@Monster@Example@MyGame....`,
  + `?fields_pack@Monster@Example@MyGame....` does conditional copy from beffer memory to the destination memory,
- iterates over memory and calls lambdas.

Internal  `std:get`:
- makes N calls of cpp17::MyGame::Example::Monster::get_field<index> + call of `StringifyField`.
Each `Monster::get_field<N>` looks like an appropriate getter.

The external  `std:get` looks much better, so I was wrong in my initial comparison.

But if add the following test case:
```cpp
  TEST_EQ(std::get<1>(monster->fields_pack()), 1);
```
we could see a bad optimization:
```asm
; 164  :   TEST_EQ(std::get<1>(monster->fields_pack()), 1);

lea	 rdx, QWORD PTR $T3[rbp-256]
mov	 rcx, rbx
call	 ?fields_pack@Monster@Example@MyGame@...
mov	 QWORD PTR [rsp+40], rdi
mov	 DWORD PTR [rsp+32], 164	; 000000a4H
lea	 r9, OFFSET FLAT:??_C@_0DB@HPEODOAI@E?3?2github?2flatbuffers?2tests?2cpp@
lea	 r8, OFFSET FLAT:??_C@_0CN@CLLIGEHH@?8std?3?3get?$DM1?$DO?$CImonster?9?$DOfields_pa@
mov	 edx, 1
movzx	 ecx, WORD PTR [rax+376]
call	 ??$TestEq@FH@@YAXFHPEBD0H0@Z ; TestEq<short,int>
```
While this code:
```cpp
TEST_EQ(monster->get_field<1>(), 1);
```
will be inlined completely and read the only `mana` field.
We need more test with `gcc` and `clang` to choose the right solution.

Reply:
I've checked the inner `std::get` with `/GL` switch (global optimization).
With this switch, all distinct calls are completely inlined as the external `std::get` does.
I'll check `gcc` and `clang` this weekend.

Reply:
Ok thanks for the analysis, I will wait to here from you on this question, and in the mean time I will implement your other suggestions.

Reply:
* Auto lambda has been removed
* Fields pack has been removed and replaced with getters which the compiler should be able to make efficient in either debug or release more.